searchlib/src/tests/attribute/extendattributes/extendattribute.cpp

```diff
@@ -185,6 +240,57 @@ void ExtendAttributeTest::testExtendRaw(AttributeVector& attr)
     EXPECT_EQ(empty, as_vector(buf));
 }
 
+void ExtendAttributeTest::testExtendTensor(AttributeVector& attr)
+{
+    std::vector<double> empty_cells{0.0, 0.0};
+    std::vector<double> spec0_dense_cells{1.0, 2.0};
+    std::vector<double> spec0_mixed_cells0{3.0, 4.0};
+    std::vector<double> spec0_mixed_cells1{5.0, 6.0};
+    bool dense = attr.getConfig().tensorType().is_dense();
+    auto* ext_attr = attr.getExtendInterface();
+    EXPECT_NE(nullptr, ext_attr);
+    auto* tensor_attr = attr.asTensorAttribute();
+    EXPECT_NE(nullptr, tensor_attr);
+    uint32_t docId(0);
+    EXPECT_EQ(0u, attr.getNumDocs());
+    attr.addDoc(docId);
```

Comment:
Consider adding a helper function to add a document. The following 3 lines are copy-pasted across all test cases.